util/secret/certificates.go

```diff
@@ -268,6 +268,16 @@ func (c Certificates) LookupOrGenerate(ctx context.Context, ctrlclient client.Cl
 	return nil
 }
 
+// HasGenerated returns true if at least one certificate was generated.
+func (c Certificates) HasGenerated() bool {
+	for _, certificate := range c {
+		if !certificate.Generated {
```

Comment:
Shouldn't this be a positive check? If any certificate is generated retun true.

Reply:
good catch!